### ISSUE TITLE
Improve S927 (C#) : For generic methods, it might be more readble to rename parameter names

### ISSUE THREAD

**Issue by stw-soloplan** [SonarSource/sonar-csharp#762]
### Description
**[Rule _S927_ (C#)](https://jira.sonarsource.com/browse/RSPEC-927)** suggests to rename a method parameter name to the name in the base class/interface declaration. Having a generic base class which accepts a parameter of the generic type `T` it sometimes makes senses to rename this parameter name for derived implementations in order to improve readability.

### Actual behavior
Assume you have an abstract parent class with the following method:
```c#
public abstract void Execute<T>(T businessObject);
```
An example implementation would be:
```c#
public override void Execute<IClient>(IClient client)
{
  if (client.Number > 0)
  {
     // do something
  }
}
```

Following the rule, this should read:

```c#
public override void Execute<IClient>(IClient businessObject)
{
 if (businessObject.Number > 0)
 {
    // do something
 }
}
```
### Expected behavior
In the above example, the parameter name `client` makes the code more readable than naming it `businessObject` which is why i think that the rule should be adjusted to incorporate such cases. Maybe this can be achieved by excluding generic method parameters from this rule.

**Comment by Evangelink** [comment]
Hi @stw-soloplan,

Thanks for the suggestion. We have discussed it internally and have checked the implementation of the FxCop we try to cover with this rule and we have decided to keep the rule the same. FxCop (CA1725) is not making exclusions for generics. Besides even if we agree on the fact the rule can be controversial we still think it makes sense for interface, generic and overrides. On your example, I think it is way cleaner because the abstract method parameter name is not the best you could have picked up.

Cheers,
Amaury
